app/jobs/govbox/check_messages_mapping_job.rb

```diff
@@ -0,0 +1,17 @@
+module Govbox
+  class CheckMessagesMappingJob < ApplicationJob
+    queue_as :default
+
+    def perform
+      unmapped_govbox_message_ids = []
+
+      Govbox::Message.find_each do |govbox_message|
+        mapped_message = ::Message.where(uuid: govbox_message.message_id)
+                           .joins(:thread).where(thread: { box_id: govbox_message.box.id }).take
+        unmapped_govbox_message_ids << govbox_message.id unless mapped_message
+      end
```

Comment:
S tymto mam 2 problemy:
1. cele by sa to asi dalo vybavit 1 selectom s left joinom (alebo subselect)
2. ako bude pocet sprav rast, tak tento job bude stale pomalsi a pomalsi. nedame tam nejaky limit, ze chceme iba spravy ktore sa stiahli mesiac dozadu?

Reply:
Viem si predstavit nieco taketo:
`unmapped_govbox_message_ids = Govbox::Message.where('delivered_at > ?', 1.months.ago).where('message_id NOT IN (?)', ::Message.where('delivered_at > ?', 1.months.ago).map(&:uuid)).map(&:id)`,
len neviem uplne ako tam zahrnut tie podmienky s tym, aby sa to vzdy checkovalo v ramci jedneho boxu. Myslel si takto nejako, ci uplne inak? @jsuchal

Reply:
NOT IN je vacsinou problem lebo to funguje ako bariera pre query planner. Skor by som cakal nieco taketo

```
SELECT * FROM govbox_messages gm
  WHERE NOT EXISTS (SELECT 1 FROM messages m WHERE m.uuid = gm.message_id LIMIT 1)
 ```
 
 alebo 
 
 ```
SELECT * FROM govbox_messages gm
  LEFT JOIN messages m ON m.uuid = gm.message_id
  WHERE m.id IS NULL
 ```
 
 A vies si tam pridat podmienky ako len chces. A do joinu aj join podmienky (napriklad pre boxy)